### DISCUSSION TITLE
Can not connect GPU with Google Colab

### DISCUSSION THREAD

**Discussion by NaziaToma** [explosion/spaCy#8747] [Help: Installation] [answered]
I have installed spacy with !pip install -U spacy[cuda112] on google colab. Then I was checking if it's connected. Here's my code:
 
import spacy
gpu = spacy.prefer_gpu()
print('GPU:', gpu)

But it's returning false which means no GPU is found. How do I solve this issue on Google Colab?

**Comment by polm** [comment, marked as answer]
On Google Colab you should **not** install cuda extras with spaCy. We don't totally understand what's going on but it seems colab has a customized version of cupy pre-installed, and you don't want to replace that with another version. 

There is an entry for this in [the FAQ](https://github.com/explosion/spaCy/discussions/8226).

**Reply by polm** [reply]
Hmmm... Actually, the behavior described in the FAQ is slightly different, and installing cuda directly gave an error. It's possible something changed on Google's end. In any case I would still suggest you try not installing cuda extras yourself and let us know if that doesn't fix it.

**Reply by NaziaToma** [reply]
I have opened a new colab and ran this code without installing. Now it works fine for me.
import spacy
gpu = spacy.prefer_gpu()
print('GPU:', gpu)

**Reply by polm** [reply]
Thanks for confirming! If you have any other issues feel free to open a new thread.
